packages/feedback2/src/modal/components/Form.tsx

```diff
@@ -0,0 +1,224 @@
+          name: retrieveStringValue(formData, 'name'),
+          email: retrieveStringValue(formData, 'email'),
+          message: retrieveStringValue(formData, 'message'),
+          attachments: attachment ? [attachment] : undefined,
+        };
+        if (!hasAllRequiredFields(data)) {
+          return;
+        }
+        try {
+          await onSubmit({ ...data, source: FEEDBACK_WIDGET_SOURCE });
+          onSubmitSuccess(data);
+        } catch (error) {
+          DEBUG_BUILD && logger.error(error);
+          setError('There was a problem submitting feedback, please wait and try again.');
+          onSubmitError(error as Error);
+        }
+      } catch {
+        // pass
+      }
+    },
+    [onSubmitSuccess, onSubmitError],
+  );
+
+  return (
+    <form class="form" onSubmit={handleSubmit}>
+      {ScreenshotInput && includeScreenshot ? <ScreenshotInput /> : null}
+
+      <div class="form__right">
+        <div class="form__top">
+          {error ? <div class="form__error-container">{error}</div> : null}
+
+          {showName ? (
+            <label for="name" class="form__label">
+              <LabelText label={nameLabel} isRequired={isNameRequired} />
+              <input
+                class="form__input"
+                defaultValue={defaultName}
+                id="name"
+                name="name"
+                placeholder={namePlaceholder}
+                required={isNameRequired}
+                type="text"
+              />
+            </label>
+          ) : (
+            <input aria-hidden value={defaultName} name="name" type="hidden" />
+          )}
+
+          {showEmail ? (
+            <label for="email" class="form__label">
+              <LabelText label={emailLabel} isRequired={isEmailRequired} />
+              <input
+                class="form__input"
+                defaultValue={defaultEmail}
+                id="email"
+                name="email"
+                placeholder={emailPlaceholder}
+                required={isEmailRequired}
+                type="text"
+              ></input>
+            </label>
+          ) : (
+            <input aria-hidden value={defaultEmail} name="email" type="hidden" />
+          )}
+
+          <label for="message" class="form__label">
+            <LabelText label={messageLabel} isRequired />
+            <textarea
+              autoFocus
+              class="form__input form__input--textarea"
+              id="message"
+              name="message"
+              placeholder={messagePlaceholder}
+              required={true}
+              rows={5}
+            />
+          </label>
+
+          {ScreenshotInput ? (
+            <label for="screenshot" class="form__label">
+              <span class="form__label__text">Screenshot</span>
+              <button
+                class="btn btn--default"
+                type="button"
+                onClick={() => {
+                  setIncludeScreeshot(prev => !prev);
```

Comment:
Does includeScreenshot update properly when you press cancel on the prompt to allow it to see the tab?

Reply:
good question, and no it doesn't!

Reply:
<img src="https://media2.giphy.com/media/gfsQffBnuc6e096brx/giphy.gif"/>

Reply:
handled it!

it's just passing the message along to the user, which won't always be the best. In the future we can get inside that `takeScreenshot().catch(...)` call and rewrite messages as needed.

![SCR-20240229-jgwc](https://github.com/getsentry/sentry-javascript/assets/187460/c54eca0d-3239-498c-848d-fd21e38619ce)